File: lingting-grpc/src/main/java/live/lingting/framework/grpc/properties/GrpcClientProperties.java
package live.lingting.framework.grpc.properties;

import live.lingting.framework.util.MdcUtils;
import lombok.Getter;
import lombok.Setter;

import java.time.Duration;

/**
 * @author lingting 2023-04-06 15:18
 */
@Getter
@Setter
public class GrpcClientProperties {

	private String host;

	private Integer port = 80;

	private String traceIdKey = MdcUtils.TRACE_ID;

	private int traceOrder = Integer.MIN_VALUE + 100;

	private boolean usePlaintext = false;

	/**
	 * 是否关闭ssl校验,仅在不使用明文时生效
	 */
	private boolean disableSsl = false;

	private boolean enableRetry = true;

	private boolean enableKeepAlive = true;

	private Duration keepAliveTime = Duration.ofMinutes(30);

	private Duration keepAliveTimeout = Duration.ofSeconds(2);

}

File: lingting-grpc/src/main/kotlin/live/lingting/framework/grpc/properties/GrpcClientProperties.kt
package live.lingting.framework.grpc.properties

import java.time.Duration
import live.lingting.framework.util.MdcUtils

/**
 * @author lingting 2023-04-06 15:18
 */
class GrpcClientProperties {
    var host: String? = null

    var port: Int = 80

    var traceIdKey: String = MdcUtils.TRACE_ID

    var traceOrder: Int = Int.MIN_VALUE + 100

    var usePlaintext: Boolean = false

    /**
     * 是否关闭ssl校验,仅在不使用明文时生效
     */
    var disableSsl: Boolean = false

    var enableRetry: Boolean = true

    var enableKeepAlive: Boolean = true

    var keepAliveTime: Duration = Duration.ofMinutes(30)

    var keepAliveTimeout: Duration = Duration.ofSeconds(2)
}
